src/sonic-config-engine/minigraph.py

```diff
@@ -101,7 +102,7 @@ def parse_png(png, hname):
                             'baud_rate': baudrate,
                             'flow_control': flowcontrol
                             }
-                    else:
+                    if startdevice.lower() == hname.lower():
```

Comment:
I do not understand why the logic is changed here. can you explain?

Reply:
When we are parsing the minigraph to generate configuration for specific ASIC, we have to make sure either startdevice or enddevice is the ASIC of interest;  as multi-asic device's minigraph will have BGP peer information of all ASIC  and external neighbors.

Reply:
reverted the change, this change was not required.